libcudacxx/include/cuda/std/__iterator/iterator_traits.h

```diff
@@ -182,36 +183,16 @@ _CCCL_TEMPLATE(class _Iter, class _Ty, class _Range)
 _CCCL_REQUIRES(_IsSame<_Iter, ::__gnu_cxx::__normal_iterator<_Ty*, _Range>>::value)
 _LIBCUDACXX_HIDE_FROM_ABI auto
   __iter_concept_fn(::__gnu_cxx::__normal_iterator<_Ty*, _Range>, __priority_tag<3>) -> contiguous_iterator_tag;
-#  elif defined(_LIBCPP_VERSION)
+#  endif // __GLIBCXX__
+#  if defined(_LIBCPP_VERSION)
 _CCCL_TEMPLATE(class _Iter, class _Ty)
 _CCCL_REQUIRES(_IsSame<_Iter, ::std::__wrap_iter<_Ty*>>::value)
 _LIBCUDACXX_HIDE_FROM_ABI auto __iter_concept_fn(::std::__wrap_iter<_Ty*>, __priority_tag<3>) -> contiguous_iterator_tag;
 #  elif defined(_MSVC_STL_VERSION) || defined(_IS_WRS)
 _CCCL_TEMPLATE(class _Iter)
-_CCCL_REQUIRES(_IsSame<_Iter, class _Iter::_Array_iterator>::value)
+_CCCL_REQUIRES((_CCCL_TRAIT(is_pointer, ::std::_Unwrapped_t<_Iter>)))
```

Comment:
I was sneaky and changed this to:
```suggestion
_CCCL_REQUIRES((_CCCL_TRAIT(is_pointer, decltype(::std::_Get_unwrapped(::std::declval<_Iter>())))))
```
before validating with older MSVC; VS2017 v15.9 doesn't have the `_Unwrapped_t` alias so we need to expand it.

Do we really need the extra set of `()` around `_CCCL_TRAIT`?

Reply:
we don't test with VS2017, so i don't think it matters. and no we don't need the extra parens.